src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -283,7 +283,7 @@ def __call__(
 
             # duplicate unconditional embeddings for each generation per prompt, using mps friendly method
             seq_len = uncond_embeddings.shape[1]
-            uncond_embeddings = uncond_embeddings.repeat(batch_size, num_images_per_prompt, 1)
```

Comment:
The change looks good to me, but I'm surprised that this doesn't work, what was the issue with it ?

Reply:
If we have three prompts and three negative prompts, batch_size would be three and we would be creating `uncond_embeddings` with shape `(9, 77, 768)` in this step. But `text_embeddings` is still `(3, 77, 768)`, and `latents_shape` in line 299 would be wrong.